### ISSUE TITLE
MaterializeMySQL will get exception when create database on cluster

### ISSUE THREAD

**Issue by Hedwiglzy** [ClickHouse/ClickHouse#19030]
**Describe the bug**
When i create MaterializeMySQL engine database on cluster,I will get the following execption,but i already set allow_experimental_database_materialize_mysql = 1 on every ClickHouse node.
If i create MaterializeMySQL engine database on single ClickHouse node,it will be successed.
```
2021.01.14 15:09:58.759295 [ 25034 ] {ee3c8887-ef0f-454b-ac10-5089e001fcae} <Error> DDLWorker: Query CREATE DATABASE test_mysql_new UUID '34f48f87-411b-4534-b64e-3bd646dd91f8' ENGINE = MaterializeMySQL('clickhouse2:3307', 'test', 'root', 'root123') wasn't finished successfully: Code: 336, e.displayText() = DB::Exception: MaterializeMySQL is an experimental database engine. Enable allow_experimental_database_materialize_mysql to use it., Stack trace (when copying this message, always include the lines below):

0. DB::InterpreterCreateQuery::createDatabase(DB::ASTCreateQuery&) @ 0xdc75c5f in /usr/bin/clickhouse
1. DB::InterpreterCreateQuery::execute() @ 0xdc81317 in /usr/bin/clickhouse
2. ? @ 0xe060347 in /usr/bin/clickhouse
3. DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, DB::Context&, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>) @ 0xe063377 in /usr/bin/clickhouse
4. DB::DDLWorker::tryExecuteQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::DDLTask const&, DB::ExecutionStatus&) @ 0xda5ec43 in /usr/bin/clickhouse
5. DB::DDLWorker::processTask(DB::DDLTask&) @ 0xda60eb9 in /usr/bin/clickhouse
6. DB::DDLWorker::enqueueTask(std::__1::unique_ptr<DB::DDLTask, std::__1::default_delete<DB::DDLTask> >) @ 0xda5f5e2 in /usr/bin/clickhouse
7. ? @ 0xda6f64d in /usr/bin/clickhouse
8. ThreadPoolImpl<ThreadFromGlobalPool>::worker(std::__1::__list_iterator<ThreadFromGlobalPool, void*>) @ 0x7dc870d in /usr/bin/clickhouse
9. ThreadFromGlobalPool::ThreadFromGlobalPool<void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()>(void&&, void ThreadPoolImpl<ThreadFromGlobalPool>::scheduleImpl<void>(std::__1::function<void ()>, int, std::__1::optional<unsigned long>)::'lambda1'()&&...)::'lambda'()::operator()() @ 0x7dcac6f in /usr/bin/clickhouse
10. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0x7dc5b3d in /usr/bin/clickhouse
11. ? @ 0x7dc96f3 in /usr/bin/clickhouse
12. start_thread @ 0x7dc5 in /usr/lib64/libpthread-2.17.so
13. __clone @ 0xf61cd in /usr/lib64/libc-2.17.so
 (version 20.12.5.14 (official build))

```
**How to reproduce**
ClickHouse server version is 20.12.5.14
MySQL server version is 8.0.22
Execute
```
create database test_materializemysql on cluster test_cluster engine = MaterializeMySQL('mysqlhost:port', 'test', 'test', 'test');
```

**Comment by tavplubix** [comment]
It's because `SET` query changes setting value in current session. Distributed DDL queries are executed in separate session and `SET` query does not affect it. You should enable `allow_experimental_database_materialize_mysql` setting globally in server config to make it work with `ON CLUSTER` queries. Example:
```
    <profiles>
            <default>
                    <allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>
                    ...
            </default>
            ...
    </profiles>
```

Another option is to configure distributed DDL queries to use separate settings profile with all settings you need:
```
    <profiles>
            <default>
                    ...
            </default>
            <dddl>
                    <allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>
            </dddl>
            ...
    </profiles>
   ...
   <distributed_ddl>
        <path>/clickhouse/task_queue/ddl</path>
        <profile>dddl</profile>
    </distributed_ddl>
```

**Comment by sliontc** [comment]
> It's because `SET` query changes setting value in current session. Distributed DDL queries are executed in separate session and `SET` query does not affect it. You should enable `allow_experimental_database_materialize_mysql` setting globally in server config to make it work with `ON CLUSTER` queries. Example:
> 
> ```
>     <profiles>
>             <default>
>                     <allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>
>                     ...
>             </default>
>             ...
>     </profiles>
> ```
> 
> Another option is to configure distributed DDL queries to use separate settings profile with all settings you need:
> 
> ```
>     <profiles>
>             <default>
>                     ...
>             </default>
>             <dddl>
>                     <allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>
>             </dddl>
>             ...
>     </profiles>
>    ...
>    <distributed_ddl>
>         <path>/clickhouse/task_queue/ddl</path>
>         <profile>dddl</profile>
>     </distributed_ddl>
> ```

I still got the error after setting the server config. Any suggestion else?
![image](https://user-images.githubusercontent.com/6959147/105567114-044d7100-5d6b-11eb-90f7-2a1609dd84ea.png)

**Comment by Hedwiglzy** [comment]
> > It's because `SET` query changes setting value in current session. Distributed DDL queries are executed in separate session and `SET` query does not affect it. You should enable `allow_experimental_database_materialize_mysql` setting globally in server config to make it work with `ON CLUSTER` queries. Example:
> > ```
> >     <profiles>
> >             <default>
> >                     <allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>
> >                     ...
> >             </default>
> >             ...
> >     </profiles>
> > ```
> > 
> > 
> > Another option is to configure distributed DDL queries to use separate settings profile with all settings you need:
> > ```
> >     <profiles>
> >             <default>
> >                     ...
> >             </default>
> >             <dddl>
> >                     <allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>
> >             </dddl>
> >             ...
> >     </profiles>
> >    ...
> >    <distributed_ddl>
> >         <path>/clickhouse/task_queue/ddl</path>
> >         <profile>dddl</profile>
> >     </distributed_ddl>
> > ```
> 
> I still got the error after setting the server config. Any suggestion else?
> ![image](https://user-images.githubusercontent.com/6959147/105567114-044d7100-5d6b-11eb-90f7-2a1609dd84ea.png)

I follow the above method to config clickhouse is OK,you need config two files,<allow_experimental_database_materialize_mysql> in user.xml,<distributed_ddl> in config.xml,did you configuration both?

**Comment by sliontc** [comment]
Yes, I figured it out that the clickhouse service have to be restarted finally.

**Comment by tavplubix** [comment]
@sliontc you should change the password. Please, be careful when posting screenshots, logs, configs and other stuff, which may contain sensitive data.

**Comment by sliontc** [comment]
> @sliontc you should change the password. Please, be careful when posting screenshots, logs, configs and other stuff, which may contain sensitive data.

OMG, could you please delete the image? I already delete my post.

**Comment by naveedmm** [comment]
Server doesn't restart after doing to suggested changes
                    `<allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>`
This is causing the restart to fail

**Comment by tavplubix** [comment]
> Server doesn't restart after doing to suggested changes `<allow_experimental_database_materialize_mysql>1</allow_experimental_database_materialize_mysql>` This is causing the restart to fail

Check server log, it contains error messages explaining why server cannot start. Most likely it's because the setting `allow_experimental_database_materialize_mysql` was renamed to `allow_experimental_database_materialized_mysql`, so server does not start due to unknown setting.
